@here/harp-mapview/lib/text/ViewState.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeometryKindSet } from "@here/harp-datasource-protocol";
+
+/**
+ * State parameters of a view that are required by the text renderer.
+ */
+export interface ViewState {
+    worldCenter: THREE.Vector3; // View's center world coordinates.
+    cameraIsMoving: boolean; // Whether view's camera is currently moving.
+    maxVisibilityDist: number; // Maximum far plane distance.
+    zoomLevel: number; // View's zoom level.
+    frameNumber: number; // Current frame number.
+    lookAtDistance: number; // Distance to the lookAt pooint.
+    isDynamic: boolean; // Whether a new frame for the view is already requested.
+    hiddenGeometryKinds: GeometryKindSet | undefined; // Kinds of geometries that are disabled.
```

Comment:
minor: `hiddenGeometryKinds?`

Reply:
See TileGeometryManager.updateTileObjectVisibility(). Objects whose kind is in the hiddenGeometryKinds set are set as invisible. Same for their labels.

Reply:
Sorry for the confusion but the `?` was not a question but an optional property. It is equivalent to `|undefined` but shorter

Reply:
done